signer/sign_state_test.go

```diff
@@ -0,0 +1,104 @@
+package signer
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSignStateCheck(t *testing.T) {
+	filepath := t.TempDir() + "/sign_state.json"
+	ss, err := LoadOrCreateSignState(filepath)
+	require.NoError(t, err)
+	require.NotNil(t, ss)
+
+	for i := 1; i < 100; i++ {
+		const jMax = 100
+		startedCh := make(chan struct{}, jMax)
+		workNowCh := make(chan struct{})
+
+		numErr, numSuccess := 0, 0
+		var mu sync.Mutex
+		i := i
+		var wg sync.WaitGroup
+		wg.Add(100)
```

Comment:
Could be more consistent about referencing the `jMax` constant here and in the counts in the final assertion. But OTOH this test may never end up changing anyway.